.github/workflows/commit-signoff-check.yml

```diff
@@ -0,0 +1,18 @@
+name: DCO commit signoff check
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  check-commit-message:
+    name: Check commit message
+    runs-on: ubuntu-latest
+    steps:
+      - name: Searching for "Signed-off"
+        uses: gsactions/commit-message-checker@v2
+        with:
+          pattern: 'Signed-off-by:'
```

Comment:
I agree with your comment about it being difficult to write a regex that matches exactly what we want, but I think we might be able to do a little better than this.  I'm not sure about JavaScript regex syntax, specifically, but what about:
```
Signed-off-by: \S+( \S+)* <\S+@\S+>
```

Each of the `\S+` expressions will match at least one non-whitespace character.  A name consists of space-delimited groups non-whitespace characters (which is general enough to allow hyphens), and at least one group is required. An e-mail address is assumed to have two parts, separated by a `@` and not containing whitespace.

You can test it here: https://regex101.com/r/pGzuoR/1

Reply:
Wow okay, that actually works a lot better than I'd thought. I'll fold this in then!

Reply:
Addressed in [aa6f52b](https://github.com/intel/p3-analysis-library/pull/12/commits/aa6f52bfc1ef14d1eb44f8ce8ded8b57b8c13d8d)